src/_pytest/main.py

```diff
@@ -167,8 +169,24 @@ def pytest_addoption(parser):
     )
 
 
+class _ConfigDeprecated(object):
+    def __init__(self, config):
+        self._config = config
```

Comment:
the setattr hook breaks this bit, use `self.__dict__["_config"]`

Reply:
ah dangit, that's what I get for writing code right as I'm pushing :laughing: